src/ngraph/op/fused/gru_cell.cpp

```diff
@@ -0,0 +1,316 @@
+
+#include <algorithm>
+#include <cmath>
+#include <functional>
+
+#include "ngraph/builder/reshape.hpp"
+#include "ngraph/builder/split.hpp"
+#include "ngraph/op/add.hpp"
+#include "ngraph/op/constant.hpp"
+#include "ngraph/op/dot.hpp"
+#include "ngraph/op/fused/gru_cell.hpp"
+#include "ngraph/shape.hpp"
+#include "ngraph/type/element_type.hpp"
+#include "ngraph/util.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+op::GRUCell::GRUCell(const shared_ptr<Node>& X,
+                     const shared_ptr<Node>& W,
+                     const shared_ptr<Node>& R,
+                     const shared_ptr<Node>& H_t,
+                     size_t hidden_size)
+    : GRUCell(X,
+              W,
+              R,
+              H_t,
+              hidden_size,
+              vector<string>{"sigmoid", "tanh"},
+              vector<float>{},
+              vector<float>{},
+              0.f,
+              false)
+{
+}
+
+op::GRUCell::GRUCell(const shared_ptr<Node>& X,
+                     const shared_ptr<Node>& W,
+                     const shared_ptr<Node>& R,
+                     const shared_ptr<Node>& H_t,
+                     size_t hidden_size,
+                     const vector<string>& activations,
+                     const vector<float>& activation_alpha,
+                     const vector<float>& activation_beta,
+                     float clip,
+                     bool linear_before_reset)
+    : FusedOp("GRUCell", {X, W, R, H_t})
+    , RNNCellBase(hidden_size, clip, activations, activation_alpha, activation_beta)
+    , m_activation_f{get_activation_function(0)}
+    , m_activation_g{get_activation_function(1)}
+    , m_linear_before_reset{linear_before_reset}
+{
+    add_default_bias_input();
+    constructor_validate_and_infer_types();
+}
+
+op::GRUCell::GRUCell(const shared_ptr<Node>& X,
+                     const shared_ptr<Node>& W,
+                     const shared_ptr<Node>& R,
+                     const shared_ptr<Node>& H_t,
+                     size_t hidden_size,
+                     const shared_ptr<Node>& B,
+                     const vector<string>& activations,
+                     const vector<float>& activation_alpha,
+                     const vector<float>& activation_beta,
+                     float clip,
+                     bool linear_before_reset)
+    : FusedOp("GRUCell", {X, W, R, H_t, B})
+    , RNNCellBase(hidden_size, clip, activations, activation_alpha, activation_beta)
+    , m_activation_f{get_activation_function(0)}
+    , m_activation_g{get_activation_function(1)}
+    , m_linear_before_reset{linear_before_reset}
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::GRUCell::pre_validate_and_infer_types()
+{
+    const auto& x_pshape = get_input_partial_shape(0);
+    const auto& w_pshape = get_input_partial_shape(1);
+    const auto& r_pshape = get_input_partial_shape(2);
+    const auto& ht_pshape = get_input_partial_shape(3);
+
+    NODE_VALIDATION_CHECK(this,
```

Comment:
Instead of throwing an error here, can't we just set the output shapes to be dynamic?